### PR TITLE
Feature/dalle

### PR DESCRIPTION
**Pull request by ethanPlumbly**

### CODE REVIEW

File: text2art/dalle/run.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        return normu
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run DALL-E and CLIP system on given text prompt",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("--prompts", nargs="+", default=[], required=True)
+    parser.add_argument("--work_dir", type=str, default="./images")
```
Comment (jplhughes) [inline text2art/dalle/run.py]: Update to --image_dir, see whats on master now to how we deal with saving intermediate images.

File: text2art/dalle/run.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        iii = perceptor.encode_image(into)
+        lat_l = 0
+        loss1 = [lat_l, 10 * -torch.cosine_similarity(t, iii).view(-1, 1).T.mean(1)]
+        loss = loss1[0] + loss1[1]
+        loss = loss.mean()
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        # if i % 100 == 0:
```
Comment (jplhughes) [inline text2art/dalle/run.py]: change 100 to be doing this `--log_every` amount of times. Also add logging for loss and timings if you want it to have parity for VQGAN output in the logs.

File: text2art/dalle/run.sh [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+echo "reason:      ${JOB_REASON}"
+echo "sge_wd:      \$(pwd)"
+echo "pstree_pid:  \$\$"
+echo
+
+echo "\$(date -u) starting \${JOB_ID}" >> ${WORK_DIR}/sge_job_id
+source $VENV/bin/activate
+
+if [[ ! -f ${WORK_DIR}/done_train ]]; then
+    python3 -m text2art.dalle.run \
```
Comment (jplhughes) [inline text2art/dalle/run.sh:83]: perhaps we use the run.sh in vqgan (but move from text2art/vqgan/run.sh --> text2art/run.sh). Then we change this line to `python3 -m text2art.$model.run`

File: text2art/dalle/run.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+from dall_e import unmap_pixels, load_model
+
+
+def displ(img, pre_scaled=True):
+    img = np.array(img)[:, :, :]
+    img = np.transpose(img, (1, 2, 0))
+    return display.Image(str(3) + ".png")
+
+
+def urlify(s):
```
Comment (jplhughes) [inline text2art/dalle/run.py]: pull from utils in vqgan, perhaps make a common util.py for these.

File: text2art/dalle/run.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
+from clip import model
+import torch
+import numpy as np
+import torchvision
+import torchvision.transforms as T
+import torchvision.transforms.functional as TF
+import argparse
+
+from IPython import display
```
Comment (jplhughes) [inline text2art/dalle/run.py]: remove

File: text2art/dalle/run.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run DALL-E and CLIP system on given text prompt",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("--prompts", nargs="+", default=[], required=True)
+    parser.add_argument("--work_dir", type=str, default="./images")
+    parser.add_argument("--image_prompts", nargs="+", default=[])
```
Comment (jplhughes) [inline text2art/dalle/run.py]: remove any arguments that aren't relevant
